docs/how_to/use_qasm2.rst

```diff
@@ -0,0 +1,174 @@
+################################
+Import and export using QASM 2.0
+################################
+
+This guide shows you how to import and export quantum circuits using ``QASM 2.0``.
+
+QASM 2.0 is short for Quantum Assembly Language version 2.0. It's a programming language specifically
+designed for representing universal quantum computing using the circuit model.
+
+You can find more information in the  `OpenQASM 2.0 repository <https://github.com/openqasm/openqasm/tree/OpenQASM2.x>`_.
+
+Design a quantum circuit
+========================
+
+In this section, you'll build a four-qubit quantum circuit using only CX (CNOT), H (Hadamard), and RZ gates.
+This circuit works with a custom quantum circuit, this is possible with the :meth:`~qiskit.QuantumCircuit.to_gate` method.
+The :meth:`~qiskit.QuantumCircuit.to_gate` method allows us to convert a part of the quantum circuit into a single gate,
+which acts like a building block encapsulating the circuit's functionality.
+
+.. plot::
+    :include-source:
+
+    from qiskit import QuantumCircuit
+    import numpy as np
+
+    custom_gate = QuantumCircuit(1) # To customize the X gate, one example is by using the H, Rz, and H gates. For this, you'll need another QuantumCircuit.
+    custom_gate.h(0)
+    custom_gate.rz(np.pi,0)
+    custom_gate.h(0)
+
+    gate = custom_gate.to_gate() # The to_gate() method converts the custom_gate() quantum circuit into a gate that you need to specify its position in your quantum circuit.
+    gate.name = 'custom X'
+
+    qc = QuantumCircuit(4,4)
+    qc.h(0)
+    qc.cx(0,1)
+    qc.cx(1,2)
+    qc.cx(2,3)
+    qc.append(gate,[1])  # To add a custom gate, you need to use the method append, where you indicate the gate and the QuantumReigsters.
```

Comment:
The comments are a bit too long, can you make them more concise? Or split them over a couple of lines?

![Screenshot 2023-08-11 at 14 21 33](https://github.com/Qiskit/qiskit-terra/assets/36071638/a6236704-bd88-41cd-8b09-3a08d358aed3)

For example:

```suggestion
    qc.append(gate,[1])  # Adds a custom gate to qubit 1
```